docs(components/link): Update for Link types and add replace/state props sections (#9971)
remix-run/react-router: docs/components/link.md

```

## `replace`

The `replace` property can be used if you'd like to replace the current entry in the history stack via [`history.replaceState`][history-replace-state] instead of the default usage of [`history.pushState`][history-push-state].

## `state`

> The `state` property can be used to set a stateful value for for the new location which is stored inside [history state][history-state]. This value can subsequently be accessed via `useLocation()`.

Reviewer: ```suggestion
The `state` property can be used to set a stateful value for the new location which is stored inside [history state][history-state]. This value can subsequently be accessed via `useLocation()`.
```

```tsx
<Link to="new-path" state={{ some: "value" }} />
```

> You can access this state value while on that location:

Reviewer: ```suggestion
You can access this state value while on the "new-path" route:
```

"that" location seemed a bit vague/confusing here.
